kmod/src/data.c

```diff
@@ -1026,6 +1032,12 @@ long scoutfs_fallocate(struct file *file, int mode, loff_t offset, loff_t len)
 		scoutfs_release_trans(sb);
 		scoutfs_inode_index_unlock(sb, &ind_locks);
 
+		/* txn couldn't meet the request. Let's try with a new txn */
+		if (ret == -ENOBUFS) {
+			scoutfs_inc_counter(sb, alloc_trans_retry);
```

Comment:
An alloc_ counter would be in alloc.c.  So if we did it this way, having one counter for the number of times we returned the errno, we'd put it up in alloc_data, not in callers.  But if we want to know what the callers are doing, then we'd have something like data_fallocate_enobufs_retry and data_write_begin_enobufs_retry counters.   Counters are cheap, maybe all of 'em?  I'll leave the specific resolution to your good judgement :).

Reply:
I thought maybe just the two, at the callsites was enough.